ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/admin/identity_management_service.proto

```diff
@@ -0,0 +1,138 @@
+  // backing participant and teh ones maintained elsewhere
+  // TODO: Check what the use case is for listing non-local parties
+  rpc ListKnownParties (ListKnownPartiesRequest) returns
+    (ListKnownPartiesResponse);
+
+  // Adds a new party to the set managed by the backing participant.
+  // Caller specifies a party identifier suggestion, the actual identifier
+  // allocated might be different and is implementation specific.
+  // daml-on-sql: suggestion's uniqueness is checked and call rejected if the
+  // identifier is already present
+  // daml-on-sawtooth: suggestion is prefixed with backing participant
+  // identifier and the composite is tested for uniqueness
+  // canton: completely different globally unique identifier is allocated.
+  // Behind the scenes calls to an internal protocol are made. As that protocol
+  // is richer than the the surface protocol, the arguments take implicit values
+  // Future extension: augment the request with additional meta-data containing
+  // human readable party name, privileges, organization etc.
+  rpc AllocateParty (AllocatePartyRequest) returns (AllocatePartyResponse);
+}
+
+message GetParticipantIdRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message GetParticipantIdResponse {
+  ParticipantDetails participant_details = 1;
+}
+
+message ParticipantDetails {
+
+  string participant_id = 1;
+}
+
+message ListKnownParticipantsRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message ListKnownParticipantsResponse {
+  repeated ParticipantDetails participant_details = 1;
+}
+
+message ListKnownPartiesRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message ListKnownPartiesResponse {
+
+  // The details of all DAML parties hosted by the participant.
+  // Required
+  repeated PartyDetails party_details = 1;
+}
+
+message PartyDetails {
+
+  // The ID of a DAML party hosted by the participant.
+  // Required
+  string party_id = 1;
+
+  // Future extension: include metadata fields. There will be two categories
+  // of such fields: common and implementation specific.
+  // CommonMetadata common_metadata = 2;
+  // bytes implementation_metadata = 3;
+  
+  // Canton-specific meta-data would contain the following:
+  // string domain = 2;
+  // string participant = 3;
+  // com.digitalasset.canton.identity.v0.ParticipantPrivilege privilege = 4;
+  // com.digitalasset.canton.identity.v0.TrustLevel trustLevel = 5;
+}
+
+message AllocatePartyRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+
+  // Party to be added to the participant.
+  // Required
+  string party_id_suggestion = 2;
```

Comment:
my suggestion is that you provide the system "party metadata" and the system replies with a "stable identifier". I.e. like in SQL:
```
CREATE TABLE parties (id INT NOT NULL AUTO_INCREMENT, name_verbose VARCHAR(255), metadata ..., UNIQUE KEY(name_verbose)) 
INSERT INTO parties (name, address) -> yields you the auto_increment INT ID. 
```

Reply:
I agree with the direction. The problem we have right now is though that our demos and tools like Navigator do rely on the stable party identifier being human readable. So I would like to keep this functionality.

What about making the suggestion optional? This way we could support both modes of operation.

Reply:
@meiersi-da yes, so nobody says that `addParty :: Text -> Text` can not just be `id`. But I think we need to set the direction now. We have oversimplified the term `Party`. Think about the following:

Company Schweizerische Bankgesellschaft is known as `SBG`. They are mentioned in all the contracts as `SBG`. Now they officially change their legal name to `UBS`. What do you do with the contracts? You need to be able to change the name without having to rewrite all the contracts.

Reply:
I think metadata might be the way to go long term, but since common_name/display_name is always going to be there, to just lift it to the top level:
`message AllocatePartyRequest {
   string ledger_id = 1;
   string common_name = 2;
}
message AllocatePartyResponse {
   string party = 1; //stable identifier, called party for consistency with other api's
   string common_name = 2;
}`